six/modules/c++/six/source/NITFWriteControl.cpp

```diff
@@ -866,11 +866,14 @@ void NITFWriteControl::save(
                 iWriter.setWriteCaching(1);
 
                 nitf::ImageSource iSource;
-                const size_t bandSize = numCols * info.getData()->getNumRows();
+                const NITFSegmentInfo segmentInfo = imageSegments[jj];
+                const size_t bandSize = numCols * segmentInfo.numRows;
```

Comment:
The NITRO comment is ambiguous here - make sure that what the `MemorySource` constructor wants is numPixels (which is what `bandSize` is) vs. number of bytes.